pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -133,16 +133,50 @@ message NamespaceOption {
     optional bool host_ipc = 3;
 }
 
+// SandboxSecurityContext holds security configuration that will be applied to a sandbox.
+message SandboxSecurityContext {
+    // The configurations for the sandbox's namespaces.
+    // This will be used only if the PodSandbox uses namespace for isolation.
+    optional NamespaceOption namespace_options = 1;
+    // Optional SELinux context to be applied. Only for sandbox. Containers should
+    // confirm to SecurityContext.SelinuxOptions.
+    optional SELinuxOption selinux_options = 2;
+    // The UID to run the entrypoint of the container process.
+    // Defaults to user specified in image metadata if unspecified.
+    // Only for sandbox. Containers should confirm to SecurityContext.RunAsUser.
+    optional int64 run_as_user = 3;
+    // Indicates that the container must run as a non-root user.
+    // If true, the runtime should validate the image at runtime to ensure that it
+    // does not run as UID 0 (root) and fail to start the container if it does.
+    // If unset or false, no such validation will be performed.
+    // Only for sandbox. Containers should confirm to SecurityContext.RunAsNonRoot.
+    optional bool run_as_non_root = 4;
+    // If set, the root filesystem of the container is read-only.
+    optional bool readonly_rootfs = 5;
```

Comment:
I'm not sure whether it is runtime-specific or not. Docker sets this for sandbox today; other runtimes may also want to perform safe behavior with docker. Since ReadOnlyRootfs is part of api.PodSecurityContext, I think CRI should expose it to runtimes and lets runtime to decide how to deal with it.

@yujuhong What do you think?